benchmarks/aggregate.py

```diff
@@ -0,0 +1,347 @@
+        results_map[timestamp][accelerator_model][dynamo][test][model_name] = {}
+      if (batch_size not in results_map[timestamp][accelerator_model][dynamo]
+          [test][model_name]):
+        results_map[timestamp][accelerator_model][dynamo][test][model_name][
+            batch_size] = {}
+      results_map[timestamp][accelerator_model][dynamo][test][model_name][
+          batch_size] = median_total_time
+
+
+def summarize_speedups(acc_map: dict[str, Any], label: str):
+  if label not in acc_map:
+    return
+  acc_map[f'{label}:gmean'] = gmean(acc_map[label])
+  for p in (5, 50, 95):
+    percentile = float(np.percentile(acc_map[label], p))
+    acc_map[f'{label}:p{p}'] = percentile
+
+
+# The speedup values are stored in acc_map[label]; the corresponding
+# model names are stored in acc_map[f'{label}:model_name'].
+def compute_speedups(acc_map: dict[str, Any], label: str, xla_label,
+                     inductor_label, test_label):
+  model_label = f'{label}:model_name'
+  if xla_label not in acc_map:
+    return
+  if inductor_label not in acc_map:
+    return
+  if (test_label not in acc_map[xla_label] or
+      test_label not in acc_map[inductor_label]):
+    return
+  for model_name, v in acc_map[xla_label][test_label].items():
+    if model_name not in acc_map[inductor_label][test_label]:
+      continue
+    speedups = []
+    # If we are running several batch sizes, keep the geomean of their speedups.
+    for batch_size in v:
+      xla_time = v[batch_size]
+      inductor_time = acc_map[inductor_label][test_label][model_name].get(
+          batch_size, None)
+      if not xla_time or not inductor_time:
+        continue
+      speedups.append(float(inductor_time) / float(xla_time))
+    if speedups:
+      if label not in acc_map:
+        acc_map[label] = []
+      acc_map[label].append(gmean(speedups))
+      if model_label not in acc_map:
+        acc_map[model_label] = []
+      acc_map[model_label].append(model_name)
+  summarize_speedups(acc_map, label)
+
+
+def process_results(results_map: dict[str, Any]):
+  for timestamp in results_map:
+    for accelerator in results_map[timestamp]:
+      acc_map = results_map[timestamp][accelerator]
+
+      compute_speedups(acc_map, 'speedups:inference', 'openxla_eval',
+                       'inductor', 'eval')
+      compute_speedups(acc_map, 'speedups:training', 'openxla', 'inductor',
+                       'train')
+
+
+def maketitle(args, title: str):
+  if args.title:
+    title += f'\n{args.title}'
+  return title
+
+
+def pr_latest(results_map: dict[str, Any], args, timestamps: list[str]):
+  label = f'speedups:{args.test}'
+  model_label = f'{label}:model_name'
+
+  for timestamp in reversed(timestamps):
+    if label not in results_map[timestamp][args.accelerator]:
+      continue
+    acc_map = results_map[timestamp][args.accelerator]
+    (speedups, model_names) = map(list, zip(*sorted(zip(acc_map[label], acc_map[model_label]))))
```

Comment:
nit: these brackets aren't needed. Think it's idiomatic not to have them

Reply:
Thanks, eventually done in https://github.com/pytorch/xla/pull/6062